### PR TITLE
Document `compute_commitment_to_aggregated_interpolation_poly()`

### PR DESCRIPTION
**Pull request by asn-d6**
I feel like `compute_commitment_to_aggregated_interpolation_poly()` is a bit complicated. I'm not sure if it can be simplified in terms of code, but IMO the documentation can be improved, so that the reader can more easily understand what's happening.

This opinionated PR adds some documentation and renames some variables. This PR should not change the logic at all.

### CODE REVIEW

File: src/eip7594/eip7594.c [hunk 1]
```diff
@@ -505,23 +505,39 @@ static C_KZG_RET compute_commitment_to_aggregated_interpolation_poly(
...
-    /* Scale each cell with the corresponding powers of the random challenge */
-    for (uint64_t i = 0; i < num_cells; i++) {
-        for (size_t j = 0; j < FIELD_ELEMENTS_PER_CELL; j++) {
+    /*
+     * Vertically collapse cells of the 2D matrix into a single array: `aggregated_column_cells`.
+     *
+     * For each provided cell, go over its field elements, and scale them by the appropriate
+     * power of r.
+     *
+     * Then aggregate all field elements on the same vertical slice into a single array.
```
Comment (jtraglia) [inline src/eip7594/eip7594.c:514]: Can we join these sentences into a single paragraph?

File: src/eip7594/eip7594.c [hunk 2]
```diff
@@ -557,26 +573,26 @@ static C_KZG_RET compute_commitment_to_aggregated_interpolation_poly(
...
         ret = fr_ifft(
             column_interpolation_poly, &aggregated_column_cells[index], FIELD_ELEMENTS_PER_CELL, s
         );
         if (ret != C_KZG_OK) goto out;
 
-        /*
-         * To unscale, divide by the coset. It's faster to multiply with the inverse. We can skip
-         * the first iteration because its dividing by one.
-         */
+        /* Now divide by the coset shift factor. */
```
Comment (jtraglia) [inline src/eip7594/eip7594.c:595]:
```suggestion
        /* Now divide by the coset shift factor */
```

**Review by jtraglia — APPROVED** [review]
LGTM, just some small nits.

File: src/eip7594/eip7594.c [hunk 3]
```diff
@@ -505,23 +505,39 @@ static C_KZG_RET compute_commitment_to_aggregated_interpolation_poly(
...
+     * Vertically collapse cells of the 2D matrix into a single array: `aggregated_column_cells`.
+     *
+     * For each provided cell, go over its field elements, and scale them by the appropriate
+     * power of r.
+     *
+     * Then aggregate all field elements on the same vertical slice into a single array.
+     */
+    for (uint64_t cell_index = 0; cell_index < num_cells; cell_index++) {
+        /* Determine which column this cell belongs to */
+        size_t column_index = cell_indices[cell_index];
```
[the remark this answers is not in the document]
Comment (asn-d6) [reply src/eip7594/eip7594.c]: Thanks! Done in 9f6d28f3

File: src/eip7594/eip7594.c (hunk 1 above)
Comment (asn-d6) [reply src/eip7594/eip7594.c]: Took the middle road in 9f6d28f3. Felt it looked better as a headline and a paragraph, but can merge the two if you prefer.

File: src/eip7594/eip7594.c (hunk 2 above)
Comment (asn-d6) [reply src/eip7594/eip7594.c]: Addressed in 9f6d28f3

File: src/eip7594/eip7594.c (hunk 1 above)
Comment (jtraglia) [reply src/eip7594/eip7594.c]: Yup that's fine. Thank you.

**Comment by asn-d6** [comment]
Thanks for review!

**Pull request merged**